### PR TITLE
Edit Some Documents

### PR DESCRIPTION
**Pull request by xiota**
* Readme: typo, other minor changes
* Crossroad: change formatting
* Debian: minor wording changes
* Docker: improve clarity
* MSYS2: typo, change formatting

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ decoder), called `libjxl`.
 JPEG XL is in the final stages of standardization and its codestream format is
 frozen.
 
-The libraries API, command line options and tools in this repository are subject
+The libraries API, command line options, and tools in this repository are subject
```
Comment (jonsneyers) [inline README.md:11]: Now that I see this, shouldn't that be `The library API` or maybe `The library's API`?

File: README.md [hunk 2]
```diff
@@ -24,9 +24,10 @@ git clone https://github.com/libjxl/libjxl.git --recursive
 ```
 
 This repository uses git submodules to handle some third party dependencies
-under `third_party/`, that's why is important to pass `--recursive`. If you
+under `third_party`, that's why is important to pass `--recursive`. If you
 didn't check out with `--recursive`, or any submodule has changed, run:
-`git submodule update --init --recursive`.
+
+    git submodule update --init --recursive
```
Comment (jonsneyers) [inline README.md:30]: add ``` and remove indent?

File: README.md (hunk 1 above)
Comment (xiota) [reply README.md]: Thanks.  I changed it to "library API".

File: README.md (hunk 2 above)
Comment (xiota) [reply README.md]: I changed it to use the tick marks.  I also edited my recent document submissions (crossroad, msys) to also use tick marks.

**Review by jonsneyers — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -49,12 +52,12 @@ in a Debian/Ubuntu based distribution run:
 sudo apt install libgif-dev libjpeg-dev libopenexr-dev libpng-dev libwebp-dev
 ```
 
-We recommend using a recent Clang compiler (version 7 or newer), for that
-install clang and set `CC` and `CXX` variables. For example, with clang-7:
+We recommend using a recent Clang compiler (at least version 6), for that
```
Comment (deymo) [inline README.md:55]: we actually recommend 7 or newer. Clang 6 "works" but it has too many SIMD-related bugs.

**Review by deymo — DISMISSED** [review]

File: README.md (hunk 3 above)
Comment (xiota) [reply README.md]: I changed it back.

**Review by deymo — APPROVED** [review]

**Pull request merged**